packages/plugin-core/src/web/views/treeView/EngineNoteProvider.ts

```diff
@@ -0,0 +1,318 @@
+            }).map((metaProps) => metaProps.id);
+
+            resolve(sortedChildren);
+          });
+        });
+
+        return;
+      } else {
+        // TODO: Change to findNotesMeta
+        this.engine.findNotes({ fname: "root" }).then((values) => {
+          const all = Promise.all(
+            values.map(async (noteProps) => {
+              this._tree[noteProps.id] = await this.createTreeNoteFromProps(
+                noteProps
+              );
+              return noteProps.id;
+            })
+          );
+
+          all.then((value) => {
+            return resolve(value);
+          });
+        });
+      }
+    });
+  }
+
+  getTreeItem(noteProps: string): TreeItem {
+    if (this._tree[noteProps]) {
+      return this._tree[noteProps];
+    } else {
+      throw new Error(`${noteProps} not found in cache!`);
+    }
+  }
+
+  dispose(): void {
+    if (this._onDidChangeTreeDataEmitter) {
+      this._onDidChangeTreeDataEmitter.dispose();
+    }
+    if (this._onEngineNoteStateChangedDisposable) {
+      this._onEngineNoteStateChangedDisposable.dispose();
+    }
+  }
+
+  private setupSubscriptions(): Disposable {
+    return this._engineEvents.onEngineNoteStateChanged((e) => {
+      e.forEach(async (noteChangeEntry) => {
+        // TODO: Add Special logic to handle deletes
+
+        this._tree[noteChangeEntry.note.id] =
+          await this.createTreeNoteFromProps(noteChangeEntry.note);
+
+        this._onDidChangeTreeDataEmitter.fire(noteChangeEntry.note.id);
+      });
+    });
+  }
+
+  private async addChildrenOfNoteToCache(
+    noteProps: NotePropsMeta
+  ): Promise<NotePropsMeta[]> {
+    return Promise.all(
+      noteProps.children.map(async (child) => {
+        if (!this._tree[child]) {
+          const props = await this.createTreeNote(child);
+          this._tree[child] = props;
+          return props.note;
+        }
+
+        return this._tree[child].note;
+      })
+    );
+  }
+
+  private async addParentOfNoteToCache(noteProps: NotePropsMeta) {
+    if (!noteProps.parent) {
+      return;
+    }
+
+    if (!this._tree[noteProps.parent]) {
+      this._tree[noteProps.parent] = await this.createTreeNote(
+        noteProps.parent
+      );
+    }
+  }
+
+  private async createTreeNote(noteId: string) {
+    const note = await this.engine.getNote(noteId);
+
+    if (!note) {
+      throw new Error(`Unable to find note ${note} for tree view!`);
+    }
+
+    return this.createTreeNoteFromProps(note);
+  }
+
+  private async createTreeNoteFromProps(note: NotePropsMeta) {
+    const collapsibleState = _.isEmpty(note.children)
+      ? TreeItemCollapsibleState.None
+      : TreeItemCollapsibleState.Collapsed;
+
+    const tn = new TreeNote(this.wsRoot, {
+      note,
+      collapsibleState,
+      labelType: this._treeViewConfig.LabelTypeSetting,
+    });
+    if (note.stub) {
+      tn.iconPath = new ThemeIcon("gist-new"); // TODO: Replace with ICONS.STUB constant
```

Comment:
why not just do this?

Reply:
I'll defer this for now - cuz constants.ts has a dependency on `engine-server`, so it'll touch quite a lot of files to make the change.